lib/core/widgets/cross_fade_button.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:deus/core/widgets/selection_button.dart';
+import 'package:deus/screens/stake_screen/stake_screen.dart';
+import 'package:deus/statics/my_colors.dart';
+import 'package:deus/statics/styles.dart';
+import 'package:flutter/material.dart';
+
+import 'filled_gradient_selection_button.dart';
+
+class CrossFadeButton extends StatelessWidget {
```

Comment:
Der Name und Speicherort suggeriert, dass es sich um ein allgemeines Widget handelt, das man für verschiedene Zwecke verwenden kann. 

Die Tatsache, dass hier festvorgegeben Strings enthalten sind und auch die Felder der Klasse bereits so benannt sind, dass es zwangsläufig nur für Approve Vorgänge genutzt werden kann, machen eine abstrakte/allgemeine Nutzung allerdings nahezu unmöglich oder zumindest erschwert möglich.

Am besten wäre es, wenn du eine allgemeine Klasse schreibst (bzw. mal in unserem theappinnovators/default_app Repo schaust, da haben wir bereits sowas, das kannst du anpassen) und dann in den einzelnen Screens nochmal die entsprechenden Abhängigkeiten der ButtonStates definierst. 

Allerdings ist das ganze, was wir hier entwickeln erstmal ein MVP. Sobald der Übersichtsscreen steht, kannst du dich dem oben genannten CrossFadeButton annehmen.
- [ ] Davor kannst du auch zur Einfachheit erstmal diese Datei kopieren
- [ ] die Klasse umbenennen (so dass erkenntlich ist, dass es nur für Screens genutzt werden kann, die was mit Approven zu tun haben) 
- [ ] und in einen neuen Unterordner core/widgets/vaults_and_staking/ verschieben. 
- [ ] ebenso verschieben von, falls vorhanden, anderen Widgets, die nicht allgemein geschrieben sind sondern nur auf Lock/Staking Screens verwendet werden können.

Wenn du später aber einen allgemeinen CrossFadeButton hast, sollten die Dateien dann darauf angepasst werden.